app/helpers/storage_resource_helper/textual_summary.rb

```diff
@@ -34,4 +35,12 @@ def textual_logical_free
   def textual_logical_total
     {:label => _("Total Size"), :value => number_to_human_size(@record.logical_total, :precision => 2)}
   end
+
+  def textual_capabilities
+    values = ""
```

Comment:
This would be better as `@record&.capabilities&.map {}.join(", ")` which will also solve your need to `.chomp` the last comma later

Reply:
will do that now.
(wouldn't need to JSON.parse before ?

Reply:
note that in the inventory we collect a **list** of hashes from the backend, so in order to save it as a hash in the jsonb field we have to do something like:
```
def parse_capabilities(capabilities)
    caps = JSON.parse(capabilities)
    caps.each do |capability|
      capability['name'] = capability.delete('abstract_capability')
    end
  end
  
  ...
  
:capabilities => {'capabilities' => parse_capabilities(resource.capability_values_json)}
```
which produces these records:
`{"href":"https://x.x.x.x/api/storage_resources/49","capabilities":{"capabilities":[{"name":"compression","uuid":"29ac2ba7-1e55-411e-9828-472c935bb882","value":"True"},{"name":"compression","uuid":"63a2dfbe-bb1b-408c-8e94-36c9d7791df2","value":"False"}]},"id":"49"}`

we considered parsing it into a hash that looks like this:
`{"compression": [{"value": "True", "uuid": "29ac2ba7-1e55-411e-9828-472c935bb882"}, {"value": "False", "uuid": "63a2dfbe-bb1b-408c-8e94-36c9d7791df2"}]}`. 
but this requires plenty of parsing at the inventory and later at the ui to actually use it, and it seems to cancel out the comfort of using a json field

Reply:
@agrare  ?

Reply:
@galoiring you can save it as an array if you want